src/scripts/breeding/Breeding.ts

```diff
@@ -237,16 +237,15 @@ class Breeding implements Feature {
     }
 
     public calculateBaseForm(pokemonName: string): string {
+        const devolution = pokemonDevolutionMap[pokemonName];
         // Base form of Pokemon depends on which regions players unlocked
-        if (!(pokemonName in pokemonDevolutionMap)) {
+        if (!devolution || PokemonHelper.calcNativeRegion(devolution) > player.highestRegion()) {
             // No devolutions at all
-            return pokemonName;
-        } else if (PokemonHelper.calcNativeRegion(pokemonDevolutionMap[pokemonName]) > player.highestRegion()) {
             // No further devolutions in current unlocked regions
             return pokemonName;
         } else {
             // Recurse onto its devolution
-            return this.calculateBaseForm(pokemonDevolutionMap[pokemonName]);
+            return this.calculateBaseForm(devolution);
```

Comment:
Why is this recursively?

if `A -> B -> C` it would make sense to only gain `A` while breeding `B` and not while breeding `C`. Although then you would also get `B` while breeding `C` and that doesn't make sense

Reply:
Wait what is preventing this from triggering on all evolved pokemons in its current form?

Reply:
I think it makes more sense that you would only get `A` while breeding, due to being the base form.
At least that’s how it works in the main series games from memory.

Reply:
The breeding map was empty due to the evolutions refactoring.

Reply:
What if we add a flag to pokémon that can only be acquired this way, and only put those in the devolutionmap

Reply:
Done, can now only receive babies via breeding/devolution map.